hoomd/mpcd/pytest/test_update.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import pytest
+import hoomd
+import numpy as np
+
+from hoomd.conftest import pickling_check
+from hoomd.logging import LoggerCategories
+from hoomd.conftest import logging_check
+
+
+@pytest.fixture
+def snap():
+    snap = hoomd.Snapshot()
+    if snap.communicator.rank == 0:
+        snap.configuration.box = [20, 20, 20, 0, 0, 0]
+        snap.particles.N = 0
+        snap.particles.types = ["A"]
+        snap.mpcd.types = ["A"]
+        snap.mpcd.mass = 1.0
+    return snap
+
+
+@pytest.fixture
+def flow():
+    flow = hoomd.mpcd.update.ReverseNonequilibriumShearFlow(
+        trigger=hoomd.trigger.Periodic(1),
+        num_swaps=1,
+        slab_width=1.0,
+        slab_separation=10.0,
+        target_momentum=2.0)
+    return flow
+
+
+class TestReverseNonequilibriumShearFlow:
+
+    def test_create(self, simulation_factory, snap, flow):
+        # before attachment
+        assert flow.trigger == hoomd.trigger.Periodic(1)
+        assert flow.num_swaps == 1
+        assert flow.slab_width == 1.0
+        assert flow.slab_separation == 10.0
+        assert flow.target_momentum == 2.0
+        with pytest.raises(hoomd.error.DataAccessError):
+            assert flow.summed_exchanged_momentum == 0.0
+
+        sim = simulation_factory(snap)
+        sim.operations.integrator = hoomd.mpcd.Integrator(dt=0.09)
+        sim.operations.add(flow)
+        sim.run(0)
+
+        # after attachment
+        assert flow.trigger == hoomd.trigger.Periodic(1)
+        assert flow.num_swaps == 1
+        assert flow.slab_width == 1.0
+        assert flow.slab_separation == 10.0
+        assert flow.target_momentum == 2.0
+        assert flow.summed_exchanged_momentum == 0.0
+
+    def test_pickling(self, simulation_factory, snap, flow):
+        pickling_check(flow)
+
+        flow.trigger = hoomd.trigger.Periodic(1)
+        flow.num_swaps = 5
+        pickling_check(flow)
+
+        sim = simulation_factory(snap)
+        sim.operations.integrator = hoomd.mpcd.Integrator(dt=0.09)
+        sim.operations.add(flow)
+        sim.run(1)
```

Comment:
You don't actually need to take a step, just call this to attach things.
```suggestion
        sim.run(0)
```